### PR TITLE
Added pcplod test (WIP)

### LINKED ISSUE #67: ensime/scala-ide support: i.e. tests with pcplod
You don't need to do anything special for it to "just work" in ensime and scala-ide, but you might find that you have bugs that break position information when the compiler enables them. I can try to help you fix it if the tests don't pass.

To ensure that you cover this use-case, please make sure to add a pcplod unit test https://github.com/ensime/pcplod

btw, I'm really looking forward to when this is ready for release, so I can use it in
https://github.com/fommil/drone-dynamic-agents/blob/master/src/main/scala/algebra.scala !

### PR DESCRIPTION
**Pull request by raulraja**
Fixes #67 

Proves generated macro code works in the scalac presentation compiler. 
Reported in #67

### CODE REVIEW

**Comment by fommil** [comment]
you have a few typos, and your sbt build is a bit messy... I got it to the point where 

```
tests/testOnly * -- -z presentation
```

fails with what looks like a real error

```
SEVERE: RangePosition(pcplodtest.scala, 248, 287, 291): value exec is not a member of pcplodtest.PcplodTestAlgebra[pcplodtest.PcplodTestAlgebra.T]
List(PcMessage(pcplodtest.scala,Error,value exec is not a member of pcplodtest.PcplodTestAlgebra[pcplodtest.PcplodTestAlgebra.T]))
[info] - should generated code works in the presentation compiler *** FAILED ***
[info]   List(PcMessage(pcplodtest.scala,Error,value exec is not a member of pcplodtest.PcplodTestAlgebra[pcplodtest.PcplodTestAlgebra.T])) was not a empty (test.scala:441)
```

and I'm sure once this is fixed up you can see how to add additional assertions

**Comment by fommil** [comment]
see https://github.com/fommil/freestyle/tree/rr-fix-67-test-pcplod

**Comment by raulraja** [comment]
@fommil Got the test to pass without any macro changes. Once I got the syntax right it was just missing the typical implicits imports.If you have a chance to take a look I'd appreciate it. Thanks!

**Comment by fommil** [comment]
sweet! Can you please add a few assertions around your code? e.g. that type at point works in a few places and so on. I assume this code example already shows that the generated code is being generated?

Incidentally, I had imagined defined an algebra using a sealed trait with classes underneath, and the methods would get generated. This is the other way round, I wasn't expecting that! It might be better this way...

**Comment by raulraja** [comment]
@fommil Added more assertions. Travis seem to fail with a RuntimeException in 2.12 because:
```
java.lang.RuntimeException: Conflicting cross-version suffixes in: org.scala-lang.modules:scala-xml
```
Could that be related to the changes in the sbt file?

**Comment by fommil** [comment]
you might want to look into adding a travis rule to only build pushes to master

File: build.sbt [hunk 1]
```diff
@@ -248,19 +188,28 @@ lazy val freestyleLoggingJS  = freestyleLogging.js
...
+    fork in Test := true,
+    javaOptions in Test ++= {
+      val options = (scalacOptions in Test).value.distinct.mkString(",")
+      val cp = (fullClasspath in Test).value.map(_.data).filter(_.exists()).distinct.mkString(",")
+      Seq(
+        s"""-Dpcplod.settings=$options""",
+        s"""-Dpcplod.classpath=$cp"""
+      )
+    },
+    dependencyOverrides += "org.scala-lang" % "scala-compiler" % scalaVersion.value
```
Comment (juanpedromoreno) [inline build.sbt]: Do we actually need this `dependencyOverrides` setting? I guess this configuration is already set up from the `projectSettings` (AutoPlugin), isn't it?

**Review by juanpedromoreno — APPROVED** [review]
I added a minor comment about a potentially duplicated settings, other than that, LGTM.

File: build.sbt (hunk 1 above)
Comment (fommil) [reply build.sbt]: yeah, that looks like a dupe.

You need way more `dependencyOverrides` than you'd think, because Ivy is dumb.

**Comment by raulraja** [comment]
@fommil Thanks for your help setting this up and thanks for pcplod, it's awesome!

**Pull request merged**